neuralop/layers/discrete_continuous_convolution.py

```diff
@@ -0,0 +1,648 @@
+                vnorm = torch.sum(psi_vals[iidx] * q[iidx])
+                psi_vals[iidx] = psi_vals[iidx] / (vnorm + eps)
+    else:
+        # pre-compute the quadrature weights
+        q = quad_weights[idx[2]].reshape(-1)
+
+        # loop through dimensions which require normalization
+        for ik in range(kernel_size):
+            for iout in range(n_out):
+                # get relevant entries
+                iidx = torch.argwhere((idx[0] == ik) & (idx[1] == iout))
+                # normalize
+                vnorm = torch.sum(psi_vals[iidx] * q[iidx])
+                psi_vals[iidx] = psi_vals[iidx] / (vnorm + eps)
+
+    return psi_vals
+
+
+def _precompute_convolution_tensor_2d(grid_in, grid_out, kernel_shape, quad_weights, normalize=True, radius_cutoff=0.01, periodic=False, transpose_normalization=False):
+    """
+    Precomputes the translated filters at positions $T^{-1}_j \omega_i = T^{-1}_j T_i \nu$. Similar to the S2 routine,
+    only that it assumes a non-periodic subset of the euclidean plane
+    """
+
+    # check that input arrays are valid point clouds in 2D
+    assert len(grid_in) == 2
+    assert len(grid_out) == 2
+    assert grid_in.shape[0] == 2
+    assert grid_out.shape[0] == 2
+
+    n_in = grid_in.shape[-1]
+    n_out = grid_out.shape[-1]
+
+    if len(kernel_shape) == 1:
+        kernel_handle = partial(_compute_support_vals_isotropic, nr=kernel_shape[0], r_cutoff=radius_cutoff)
+    elif len(kernel_shape) == 2:
+        kernel_handle = partial(_compute_support_vals_anisotropic, nr=kernel_shape[0], nphi=kernel_shape[1], r_cutoff=radius_cutoff)
+    else:
+        raise ValueError("kernel_shape should be either one- or two-dimensional.")
+
+    grid_in = grid_in.reshape(2, 1, n_in)
+    grid_out = grid_out.reshape(2, n_out, 1)
+
+    diffs = grid_in - grid_out
+    if periodic:
+        periodic_diffs = torch.where(diffs > 0.0, diffs - 1, diffs + 1)
+        diffs = torch.where(diffs.abs() < periodic_diffs.abs(), diffs, periodic_diffs)
+
+    r = torch.sqrt(diffs[0] ** 2 + diffs[1] ** 2)
+    phi = torch.arctan2(diffs[1], diffs[0]) + torch.pi
+
+    idx, vals = kernel_handle(r, phi)
+
+    idx = idx.permute(1, 0)
+
+    if normalize:
+        vals = _normalize_convolution_tensor_2d(idx, vals, grid_in, grid_out, kernel_shape, quad_weights, transpose_normalization=transpose_normalization)
+
+    return idx, vals
+
+
+class DiscreteContinuousConv2d(DiscreteContinuousConv):
+    """
+    Discrete-continuous convolutions (DISCO) on arbitrary 2d grids as implemented for [1]. To evaluate continuous convolutions on a computer, they can be evaluated semi-discretely, where the translation operation is performet continuously, and the qudrature/projection is performed discretely on a grid [2]. They are the main building blocks for local Neural Operators [1]. Forward call expects an input of shape batch_size x in_channels x n_in.
```

Comment:
```suggestion
    Discrete-continuous convolutions (DISCO) on arbitrary 2d grids as implemented for [1]. To evaluate continuous convolutions on a computer, they can be evaluated semi-discretely, where the translation operation is performed continuously, and the quadrature/projection is performed discretely on a grid [2]. They are the main building blocks for local Neural Operators [1]. Forward call expects an input of shape batch_size x in_channels x n_in.
```
Just taking a look around, sorry to nitpick :sweat_smile: